foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/spi/AnalyticsLoader.java

```diff
@@ -60,17 +61,19 @@
     private final CacheManager cacheManager;
     private final ObjectMapper objectMapper;
     private final EmailConfig emailConfig;
+    private EmailClient emailClient;
 
     public AnalyticsLoader(TableMetadataManager tableMetadataManager, DataStore dataStore, QueryStore queryStore,
                            ElasticsearchConnection elasticsearchConnection, CacheManager cacheManager, ObjectMapper objectMapper,
-                           EmailConfig emailConfig) {
+                           EmailConfig emailConfig, EmailClient emailClient) {
```

Comment:
Why is analyticsloader and actionexecutor interface polluted with email stuff? Remove both and create a listener interface for exception. Implement that to send email. Add loader to register listeners to action.

Reply:
Will make these changes